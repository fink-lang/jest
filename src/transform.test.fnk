{describe, it, expect, to_match_snapshot, to_throw} = import '.'
{process} = import './transform'


describe process, fn:

  it 'compiles fink', fn:
    {code, map} = process
      'foo = 1234 or spam'
      '/temp/src/foobar.fnk'
      {rootDir: '/temp/src'}

    expect
      {code, map}
      to_match_snapshot


  it 'compiles fink with cov', fn:
    {code, map} = process
      'foo = 1234 or spam'
      '/temp/src/foobar.fnk'
      {rootDir: '/temp/src'}
      {instrument: true}

    expect
      {code, map}
      to_match_snapshot


  it 'transforms mock imported from @fink/jest as jest.mock', fn:
    {code, map} = process
      `
        {it, mock, mock_fn} = import '@fink/jest'

        mock './transform.fnk', fn: dict:
          foobar: mock_fn _

        foo = fn mock:
          mock 'not jest.mock'
      `
      '/temp/src/foobar.fnk'
      {rootDir: '/temp/src'}

    expect
      {code, map}
      to_match_snapshot


  it 'ignores transforming mock not imported from @fink/jest', fn:
    {code, map} = process
      `
        {mock} = import './foobar.fnk'

        mock 'not a jest mock'
      `
      '/temp/src/foobar.fnk'
      {rootDir: '/temp/src'}

    expect
      {code, map}
      to_match_snapshot


  it 'ignores transforming mock not imported', fn:
    {code, map} = process
      `
        {mock} = spam 'ni'

        mock 'not a jest mock'
      `
      '/temp/src/foobar.fnk'
      {rootDir: '/temp/src'}

    expect
      {code, map}
      to_match_snapshot


  it 'handles larix errors', fn:
    expect
      fn: process '123 foo', 'foobar.fnk', {rootDir: './src'}
      to_throw '
        foobar.fnk:1:4
        1| 123 foo
               ^

        Expected ',' or indented new line or Symbol(end).'


  it 'handles loxia errors', fn:
    expect
      fn: process '123 = foo', 'foobar.fnk', {rootDir: './src'}
      to_throw `
        foobar.fnk:1:0
        1| 123 = foo
           ^

        Unable to transform 'assign ='`
