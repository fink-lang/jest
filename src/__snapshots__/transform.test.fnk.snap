// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process compiles fink 1`] = `
Object {
  "code": "const foo = 1234 || spam;
Object.assign(module.exports, {
  foo
});",
  "map": Object {
    "mappings": "MAAA,G,GAAM,QAAQ,I",
    "names": Array [],
    "sources": Array [
      "src/foobar.fnk",
    ],
    "sourcesContent": Array [
      "foo = 1234 || spam",
    ],
    "version": 3,
  },
}
`;

exports[`process compiles fink with cov 1`] = `
Object {
  "code": "/* istanbul ignore next */
function cov_1eidqdiw3e() {
  var path = \\"/Users/jan/Develop/fink/jest/src/foobar.fnk\\";
  var hash = \\"6690b9df05185ac70af1b85723cc7aadfd6799be\\";
  var global = new Function(\\"return this\\")();
  var gcv = \\"__coverage__\\";
  var coverageData = {
    path: \\"/Users/jan/Develop/fink/jest/src/foobar.fnk\\",
    statementMap: {
      \\"0\\": {
        start: {
          line: 1,
          column: 12
        },
        end: {
          line: 1,
          column: 24
        }
      },
      \\"1\\": {
        start: {
          line: 2,
          column: 0
        },
        end: {
          line: 4,
          column: 3
        }
      }
    },
    fnMap: {},
    branchMap: {
      \\"0\\": {
        loc: {
          start: {
            line: 1,
            column: 12
          },
          end: {
            line: 1,
            column: 24
          }
        },
        type: \\"binary-expr\\",
        locations: [{
          start: {
            line: 1,
            column: 12
          },
          end: {
            line: 1,
            column: 16
          }
        }, {
          start: {
            line: 1,
            column: 20
          },
          end: {
            line: 1,
            column: 24
          }
        }],
        line: 1
      }
    },
    s: {
      \\"0\\": 0,
      \\"1\\": 0
    },
    f: {},
    b: {
      \\"0\\": [0, 0]
    },
    inputSourceMap: {
      version: 3,
      sources: [\\"./src/foobar.fnk\\"],
      names: [],
      mappings: \\"MAAA,G,GAAM,QAAQ,I\\",
      sourcesContent: [\\"foo = 1234 || spam\\"]
    },
    _coverageSchema: \\"1a1c01bbd47fc00a2c39e90264f33305004495a9\\",
    hash: \\"6690b9df05185ac70af1b85723cc7aadfd6799be\\"
  };
  var coverage = global[gcv] || (global[gcv] = {});

  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }

  var actualCoverage = coverage[path];

  cov_1eidqdiw3e = function () {
    return actualCoverage;
  };

  return actualCoverage;
}

cov_1eidqdiw3e();
const foo =
/* istanbul ignore next */
(cov_1eidqdiw3e().s[0]++,
/* istanbul ignore next */
(cov_1eidqdiw3e().b[0][0]++, 1234) ||
/* istanbul ignore next */
(cov_1eidqdiw3e().b[0][1]++, spam));

/* istanbul ignore next */
cov_1eidqdiw3e().s[1]++;
Object.assign(module.exports, {
  foo
});",
  "map": Object {
    "mappings": ";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;MAAA,G;;;AAAM;AAAA;AAAA;AAAA,6BAAQ,IAAR,C",
    "names": Array [],
    "sources": Array [
      "src/foobar.fnk",
    ],
    "sourcesContent": Array [
      "foo = 1234 || spam",
    ],
    "version": 3,
  },
}
`;
